### PR TITLE
Adding the missing defined for HPX_HAVE_DEPRECATION_WARNINGS

### PR DESCRIPTION
**Pull request by shahrzad**

### CODE REVIEW

**Comment by hkaiser** [comment]
add to whitelist

**Review by hkaiser — APPROVED** [review]
LGTM, thanks!

**Review by aurianer — APPROVED** [review]
Thanks!

File: libs/core/config/include/hpx/config/attributes.hpp [hunk 1]
```diff
@@ -63,7 +63,7 @@
 
 ///////////////////////////////////////////////////////////////////////////////
 // handle [[deprecated]]
-#if HPX_HAVE_DEPRECATION_WARNINGS && !defined(HPX_INTEL_VERSION)
+#if defined(HPX_HAVE_DEPRECATION_WARNINGS) && !defined(HPX_INTEL_VERSION)
```
Comment (msimberg) [inline libs/core/config/include/hpx/config/attributes.hpp:66]: Um, I think it was not using `defined` on purpose. `HPX_HAVE_DEPRECATION_WARNINGS` is added with `hpx_add_config_cond_define` which means that one can add `-DHPX_HAVE_DEPRECATION_WARNINGS=0` to manually disable deprecation warnings even if HPX was built with deprecation warnings on.

Did not having `defined(HPX_HAVE_DEPRECATION_WARNINGS)` cause problems?

Comment (hkaiser) [reply libs/core/config/include/hpx/config/attributes.hpp:66]: @msimberg You are certainly right. I asked @shahrzad to create this PR but I missed that we left it that way on purpose :/ Should we roll it back?

Comment (msimberg) [reply libs/core/config/include/hpx/config/attributes.hpp:66]: I would, yes, unless this really fixed something that was broken? And if it fixed something that was broken, the fix should probably look different.

Comment (hkaiser) [reply libs/core/config/include/hpx/config/attributes.hpp:66]: See #5430

**Pull request merged**